src/gui/connectionWidget.h

```diff
@@ -29,20 +29,76 @@
 #define CONNECTIONWIDGET_H_
 
 #include <QWidget>
+#include "../deviceCommunication/commMaster.h"
 
 namespace Ui {
 class ConnectionWidget;
 }
 
+/**
+ * @brief Class to display informations about a single connection
+ *
+ * This class can display information extracted from the data stream to the user.
+ * It also enables interaction with a device; e.g. change the frequency or
+ * disconnect from a device.
+ *
+ * @todo Add method to update the remaining data (battery, frequency...)
+ * @todo Differentiate between USB and BLE regarding supported frequencies
+ *
+ */
 class ConnectionWidget : public QWidget {
     Q_OBJECT
 
    public:
+
+    /**
+     * @brief Constructor of the class
+     * 
+     * The constructor initialize the frequency selector with the available
+     * frequency values.
+     * 
+     * \warning Call the method addCommunication directly after construction
+     * to initialize the communication pointer.
+     *
+     * @param parent Pointer to parent widget, used for parent/child relation of qt
+     */
     ConnectionWidget(QWidget* parent = nullptr);
     ~ConnectionWidget();
 
+    /**
+     * @brief Add the pointer to the main instance of the CommMaster class
+     *
+     * This pointer is used to connected a new device and get all available devices.
+     * The reason this is done in a method and not inside the constructor
+     * is the autogeneration of ui_dialogconnect.ui witch requires a constructor
+     * with a single parameter QWidget*.
+     * 
+     * \warning Call this method directly after construction to initialize the 
+     * communication pointer.
```

Comment:
```suggestion
     * @brief Set the `CommMaster` used by this `ConnectionWidget`.
     *
     * This pointer is used to connect new devices and get all available devices.
     * This setter exists because this widget is set up automatically through
     * auto-generated (code as defined in `ui_dialogconnect.ui`) which requires
     * a constructor with a single `QWidget*` argument.
     * 
     * \note Call this method directly after construction to assign the 
     * communication master.
```